b-micromax-md3-pc/exporter.py

```diff
@@ -444,15 +444,74 @@ def _do_start_raster_scan(
     def _do_start_scan_ex(
         self,
         _frame_id,
-        _start_angle,
-        _scan_range,
+        start_angle,
+        scan_range,
         _exposure_time,
         _number_of_passes,
     ) -> int:
-        return self._add_task("Start SCAN", 3.2)
-
-    def _do_start_scan_4d_ex(self, *_):
-        return self._add_task("Start 4D-SCAN", 3.2)
+        TASK_DURATION = 3.2 
+        async def start_scan():
+            self.write_attribute("FastShutterIsOpen", True)
+            state_attr = "OmegaState"
+            pos_attr = "OmegaPosition"
+            step_size = float(scan_range) / MOTOR_STEPS
+            self.write_attribute(state_attr, "Moving")
+            for _ in range(MOTOR_STEPS):
+                current_pos = self._attrs[pos_attr].val
+                log(f"INFO: {current_pos=} for {pos_attr=}, will move by {step_size=}")
+                self.write_attribute(pos_attr, current_pos + step_size)
```

Comment:
The `Omega` value needs to wrap around 0..360 range. See for example how it's done here:

https://github.com/elmjag/micromax/blob/9332db83cca50ba87a0e5027c6310b31fbad24ea/b-micromax-md3-pc/exporter.py#L636

I think it's time to introduce a dedicated function for this.

```
def _wrap_omega_position(new_omega_position: float) -> float:
  """Keep Omega rotation angle within 0..360 range"""
  return new_omega_position % 360.0
```